dev/create_worker.sh

```diff
@@ -1,28 +1,112 @@
-echo "Retrieving access token"
+#!/bin/bash
+
+# Call it once to create a `test_worker`:
+# - retrieve an admin token
+# - create the `test_worker`` user
+# - create the associated worker object
+# - upload a test public key.
+#
+# To be used to have a "real" test worker for local development, typically to start
+# a worker manager or a task manager or simply assign tasks to a worker in the UI/API
+
+set -e
+
+ZIMFARM_BASE_API_URL="http://localhost:8004/v2"
+
+die() {
+    echo "ERROR: $1" >&2
+    exit 1
+}
+
+check_http_code() {
+    local http_code="$1"
+    local response="$2"
+
+    if [ "$http_code" -ge 200 ] && [ "$http_code" -lt 300 ]; then
+	:
+    else
+	error_msg=$(echo "$response" | jq -r '.errors // .message // .detail // "Unknown error"' 2>/dev/null || echo "HTTP $http_code")
+	die "Could not checkin worker: ${error_msg}"
+    fi
+}
+
+echo "Retrieving admin access token"
 
 ZF_ADMIN_TOKEN="$(curl -s -X 'POST' \
-    'http://localhost:8002/v2/auth/authorize' \
+    "${ZIMFARM_BASE_API_URL}/auth/authorize" \
     -H 'accept: application/json' \
     -H 'Content-Type: application/json' \
-    -d '{"username":"admin","password":"admin"}' \
+    -d '{"username": "admin", "password": "admin"}' \
     | jq -r '.access_token')"
 
+if [ -z "$ZF_ADMIN_TOKEN" ] || [ "$ZF_ADMIN_TOKEN" = "null" ]; then
+    die "Failed to retrieve admin access token"
+fi
 
-echo "Worker check-in (will create if missing)"
+echo "Create test_worker user"
 
-curl -s -X 'PUT' \
-  'http://localhost:8002/v2/workers/worker/check-in' \
+curl -s -X 'POST' \
+  "${ZIMFARM_BASE_API_URL}/users" \
   -H 'accept: */*' \
   -H 'Content-Type: application/json' \
   -H "Authorization: Bearer $ZF_ADMIN_TOKEN" \
   -d '{
-  "username": "admin",
-  "cpu": 3,
-  "memory": 1024,
-  "disk": 0,
+  "role":"worker",
+  "username": "test_worker",
+  "email":"test_worker@acme.com",
+  "password":"test_worker"
+}'
+
+echo "Retrieving test_worker access token"
+
+ZF_USER_TOKEN="$(curl -s -X 'POST' \
+    "${ZIMFARM_BASE_API_URL}/auth/authorize" \
+    -H 'accept: application/json' \
+    -H 'Content-Type: application/json' \
+    -d '{"username": "test_worker", "password": "test_worker"}' \
+    | jq -r '.access_token')"
+
+if [ -z "$ZF_USER_TOKEN" ] || [ "$ZF_USER_TOKEN" = "null" ]; then
+    die "Failed to retrieve worker access token"
+fi
+
+response=$(curl -s -w "\n%{http_code}" -X 'PUT' \
+  "${ZIMFARM_BASE_API_URL}/workers/test_worker/check-in" \
+  -H 'accept: */*' \
+  -H 'Content-Type: application/json' \
+  -H "Authorization: Bearer $ZF_USER_TOKEN" \
+  -d '{
+  "username": "test_worker",
+  "cpu": 1,
```

Comment:
We should assign more CPU by default. Otherwise zimit-frontend fails to assign the task to the worker, it requires 3 ATM.